### PR TITLE
Issue 521: Docker Support

### PR DESCRIPTION
**Pull request by Hzaakk**
Adding `Dockerfile`s and a `docker-compose.yml`

How to use:
1. Generate configuration (`dev_settings.py`): have Python with the `jinja2` and `pyyaml` libraries installed, and then run `bootstrap/development/gen_config.sh`
2. Build Images: In the base directory, run `docker build . -t coldfront` and `docker build . -f Dockerfile.db -t coldfront_db`
3. To run: In the base directory, run `docker compose up`
4. To enter the container (similar to `vagrant ssh`): run `docker exec -it coldfront-coldfront-1 bash`
5. To load a database backup: run `bootstrap/development/docker_load_database_backup.sh ${DB_NAME} ${PATH_TO_DUMP}`
6. To start from scratch (delete volumes): In the base directory, run `docker compose down --volumes`

### CODE REVIEW

**Comment by matthew-li** [comment]
@Hzaakk

@ViRu-ThE-ViRuS wants to know how big the image sizes are, due to limited disk space--can you list those here?

**Comment by ViRu-ThE-ViRuS** [comment]
> @ViRu-ThE-ViRuS wants to know how big the image sizes are, due to limited disk space--can you list those here?
image size on Mac: 1.22GB

File: Dockerfile [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
+RUN pip install jinja2 pyyaml
+
+# mybrc or mylrc
+ARG portal="mybrc"
+RUN mkdir -p /var/log/user_portals/cf_${portal} \
+ && touch /var/log/user_portals/cf_${portal}/cf_${portal}_{portal,api}.log \
+ && chmod 775 /var/log/user_portals/cf_${portal} \
+ && chmod 664 /var/log/user_portals/cf_${portal}/cf_${portal}_{portal,api}.log
+
+ARG base_dir="/vagrant/coldfront_app"
```
Comment (matthew-li) [inline Dockerfile:18]: This path is hardcoded in a volume in `docker-compose.yml` but configurable here. I'd say we can keep it hardcoded.

**Review by matthew-li — CHANGES_REQUESTED** [review]
Great job--this worked for me. I left some minor comments.

**Pull request merged**